src/test/kotlin/org/jetbrains/research/transformations/util/ParserSetup.kt

```diff
@@ -3,28 +3,56 @@ package org.jetbrains.research.transformations.util
 import org.apache.commons.io.FileUtils
 import org.jetbrains.research.transformations.util.ParserSetup.checkSetup
 import java.io.File
+import java.io.InputStream
+import java.net.URL
+import java.nio.file.Files
+import java.nio.file.Paths
+import java.nio.file.StandardCopyOption
 import java.util.logging.Logger
 
+
 /**
  * [ParserSetup] class is created for parser setup before using Gumtree with python code.
  * Also [checkSetup] should be called before running tests.
  */
 object ParserSetup {
-    private val LOG = Logger.getLogger(javaClass.name)
+    private val logger = Logger.getLogger(javaClass.name)
+
+    private const val PARSER_REPOSITORY_ZIP_URL = "https://github.com/elena-lyulina/pythonparser/archive/master.zip"
+    private const val PARSER_ZIP_NAME = "master.zip"
+
+    // Relative path in the parser repository
+    private const val PARSER_RELATIVE_PATH = "pythonparser-master/src/main/python/pythonparser/pythonparser_3.py"
+
     private const val PARSER_NAME = "pythonparser"
-    private val TARGET_PATH = "${getTmpPath()}/$PARSER_NAME"
+    private val TARGET_PARSER_PATH = "${getTmpPath()}/$PARSER_NAME"
 
     /**
-     * Puts parser file to the target path.
-     * @param targetPath - path to the final file destination,
-     * where it will be putted.
+     * Get parser repository path in this project in the resources folder
      */
-    private fun putParserToTargetPath(targetPath: String= TARGET_PATH) {
-
-        LOG.info("Putting parser into $targetPath")
-        val pythonparserFile = File(javaClass.getResource("$PARSER_NAME.py").path)
-        pythonparserFile.copyTo(File(targetPath), overwrite = true)
+    private fun getParserRepositoryPath(): String {
+        val zipFilePath = Paths.get(javaClass.getResource(PARSER_ZIP_NAME).path)
+        return zipFilePath.parent.toString()
+    }
 
+    /**
+     * Unzip parser repository
+     * @param toUpdate - it is true, if a new version of the parser needs to download
+     */
+    private fun unzipParserRepo(toUpdate: Boolean = true) {
+        val zipFilePath = Paths.get(javaClass.getResource(PARSER_ZIP_NAME).path)
+        if (toUpdate) {
+            logger.info("Updating the current master zip")
+            val file: InputStream = URL(PARSER_REPOSITORY_ZIP_URL).openStream()
+            Files.copy(file, zipFilePath, StandardCopyOption.REPLACE_EXISTING)
+        }
+        logger.info("Unzipping the folder with the repository")
+        ProcessBuilder()
+            .command("unzip", "-o", zipFilePath.toString(), "-d", getParserRepositoryPath())
```

Comment:
For some reason, it unpacks the zip file into the **build** folder, was it intended?
`zipFilePath.toString()` is ast-canonicalization/**build**/resources/test/org/jetbrains/research/transformations/util/master.zip and `getParserRepositoryPath()` returns ast-canonicalization/**build**/resources/test/org/jetbrains/research/transformations/util

Reply:
Yes, because "unzip" command called every time tests run. In the resource folder, I sore only a zip file. I would like to ask you, maybe should I use something like this:
```
val zipFile = ZipFile(zipFilePath.toString())
zipFile.extractAll(getParserRepositoryPath())
```
instead of `ProcessBuilder`?

Reply:
I think `extractAll` looks better, so yes